tests/test_initialization.py

```diff
@@ -1763,6 +1765,471 @@ def new_state_dict():
     assert any(missing_key in str(w.message) for w in recwarn.list)
 
 
+class TestCordaInitialization:
+    """Test class to check the initialization of CorDA adapters."""
+
+    torch_device = infer_device()
+
+    def get_uniform(self, amin, amax, size=(10000,)):
+        unif = torch.distributions.uniform.Uniform(amin, amax)
+        samples = unif.sample(size)
+        return samples
+
+    def get_normal(self, mean, std, size=(10000,)):
+        normal = torch.distributions.normal.Normal(mean, std)
+        samples = normal.sample(size)
+        return samples
+
+    def get_model(self):
+        class MyModule(nn.Module):
+            def __init__(self):
+                super().__init__()
+                # choose a large weight so that averages are close to expected values
+                self.linear = nn.Linear(1000, 1000)
+
+            def forward(self, x):
+                return self.linear(x)
+
+        return MyModule().eval().to(self.torch_device)
+
+    @pytest.fixture
+    def data(self):
+        # larger data is required to pass KPM test
+        return torch.rand(1000, 1000).to(self.torch_device)
+
+    @pytest.mark.parametrize("corda_method", ("ipm", "kpm"))
+    def test_lora_corda_linear_init_default(self, data, tmp_path, corda_method):
```

Comment:
Checking this test, we're always asserting that the CorDA outputs are identical to the output of the base model. I know that this is expected, but it makes the test a bit weak, since if CorDA just did nothing, the test would still pass. Is there an additional check that could be performed to ensure that `preprocess_corda` is not just a no-op?

Reply:
You're right, I've modified the tests so that it doubles LoRA weight, and check if output is different from original output. This way if CorDA is no-op, the weights are initialized zero and the tests would not pass.